File: src/CorporateMessengerWithAgile/Application/Dto/TaskItemDto.cs
namespace Application.Dto
{
    public class TaskItemDto : BaseDto
    {
        public Guid ProjectId { get; set; }
        public Guid AuthorId { get; set; }
        public Guid ResponsibleId { get; set; }
        public Guid? SprintWithLastMentionId { get; set; }
        public Guid? ParentTaskId { get; set; }
        public IReadOnlyList<Guid> SubtaskIds { get; set; } = null!;
        public string Title { get; set; } = null!;
        public string Description { get; set; } = null!;
        public int Priority { get; set; }
        public int Complexity { get; set; }
        public DateTime Deadline { get; set; }
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/TaskItemDto.ts
// src/models/entity/TaskItemDto.ts
import { BaseDto } from "./BaseDto";
import { Guid } from "../Guid";

export class TaskItemDto extends BaseDto {
    public projectId: Guid;
    public authorId: Guid;
    public responsibleId: Guid;
    public sprintWithLastMentionId?: Guid;
    public parentTaskId?: Guid;
    public subtaskIds: Guid[];
    public title: string;
    public description: string;
    public priority: number;
    public complexity: number;
    public deadline: Date;

    constructor(
        id: Guid,
        projectId: Guid,
        authorId: Guid,
        responsibleId: Guid,
        title: string,
        description: string,
        priority: number,
        complexity: number,
        deadline: Date,
        sprintWithLastMentionId?: Guid,
        parentTaskId?: Guid,
        subtaskIds: Guid[] = []
    ) {
        super(id);
        this.projectId = projectId;
        this.authorId = authorId;
        this.responsibleId = responsibleId;
        this.sprintWithLastMentionId = sprintWithLastMentionId;
        this.parentTaskId = parentTaskId;
        this.subtaskIds = subtaskIds;
        this.title = title;
        this.description = description;
        this.priority = priority;
        this.complexity = complexity;
        this.deadline = deadline;
    }
}
